ArjunaJTA/jta/classes/com/arjuna/ats/internal/jta/transaction/arjunacore/jca/XATerminatorImple.java

```diff
@@ -319,12 +324,20 @@ public int prepare (Xid xid) throws XAException
         case XAResource.TMSTARTRSCAN: // check the object store
             if (_recoveryStarted)
                 throw new XAException(XAException.XAER_PROTO);
-            else
+            else {
                 _recoveryStarted = true;
+                if (XARecoveryModule.getRegisteredXARecoveryModule() != null) {
+                    XARecoveryModule.getRegisteredXARecoveryModule().periodicWorkFirstPass();
```

Comment:
Remark: if periodicWorkFirstPass() is already running this call will wait and then trigger another first pass which is wasteful of CPU cycles. Prior to your changes repeating the first pass call would have been skipped since we had the following check in first pass (which you removed):
```java
if (!getScanState().equals(ScanStates.IDLE))
```